feat: upgrade to hydromt v1

LuukBlom opened the pull request:
## Issue addressed

Fixes #1 

Requires hydromt 1.3.0 to be released. 
TODO: update hydromt version in `pyproject.toml` to 1.3

## Explanation
Implemented hydromt v1 according to migration guide: 
- components
- data catalog format

hydromt deprecated .ini files for workflows, so I updated docs. There are no example workflow .ini files to refactor.

I also renamed the main config `.ini` file to a `.toml` since `.toml` is the industry standard and `.ini` is outdated. @daanrooze, please let me know if this is okay or not. There are no differences in the contents at the moment, the file extension is the only difference.

## Checklist
- [ ] Updated tests or added new tests
- [ ] Branch is up to date with `main`
- [ ] Tests & pre-commit hooks pass
- [ ] Updated documentation if needed
- [ ] Updated changelog.rst if needed

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

LuukBlom committed:
updated to hydromt v1. Almost done. TODO: dynamic buffer arg in get_geodataframe in setup_landuse

LuukBlom committed:
- landuse: updated `landuse_table` and `landuse_from_osm` functions
- use Model.test_equal for equality in testing
- Regenerate the test model

LuukBlom committed:
fix defaults in config

LuukBlom committed:
fix imports

LuukBlom committed:
Merge branch 'master' into feat/hydromt-v1

LuukBlom committed:
Merge branch 'master' into feat/hydromt-v1

LuukBlom committed:
Merge branch 'master' into feat/hydromt-v1

LuukBlom committed:
python versions

LuukBlom committed:
replace .ini with .toml

LuukBlom committed:
Merge branch 'master' into feat/hydromt-v1

LuukBlom committed:
added repo secret and updated workflow for test coverage

deltamarnix commented on .github/workflows/tests.yml:
```diff
@@ -38,11 +38,13 @@ jobs:
           flags: unittests
-          name: codecov-umbrella
           fail_ci_if_error: true
+          disable_search: true
```
I see codecov was already here, but we can actually use SonarQube for this. How much work would it be to change it to the SonarQube action? There is a Deltares-Research where you can add the project: https://sonarcloud.io/organizations/deltares-research/projects

LuukBlom committed:
update hydromt dependency to git @ main, temporary until 1.3.0 is released

deltamarnix commented on hydromt_uwbm/uwbm.py:
```diff
@@ -64,20 +50,41 @@ def __init__(
+            ).as_posix(),
+        )
+        self.forcing = UWBMForcingComponent(self, filename="input/{name}.csv")
+        self.geoms = GeomsComponent(
```
For HydroMT we decided, although not required, that we also created empty classes with inheritance. I'm okay with using the default components if there is nothing to inherit, but I did want to mention it.

deltamarnix reviewed (commented):
I wonder if ConfigComponent can be implemented in a quicker way. It looks like the dictionary is there, but just before writing everything is dumped into a pydantic object that also does validation.

Wouldn't it be easier to not inherit from ConfigComponent, but just add the pydantic values directly to the class? And no more get, set functions, instead just use the attributes from pydantic like starttime, endtime?

daanrooze commented on tests/data/Athens_Votris_model/input/neighbourhood_params.toml:
```diff
@@ -0,0 +1,104 @@
+
+title = "neighbourhood config"  # Title of the configuration
+name = "Athens_Votris"          # Name of the simulation
+starttime = 2020-03-01 00:00:00 # Simulation start time in format YYYY-MM-DD HH:MM:SS
```
starttime and endtime do not have to be written to .ini file

deltamarnix commented on docs/user_guide/uwbm.qmd:
````diff
@@ -59,7 +59,7 @@ This plugin allows to build from scratch a complete model from available data.
 4. **Build** your model using the CLI or Python interface.
 
 ```
-    hydromt build uwbm path/to/built_model -r "{'bbox': [xmin, ymin, xmax, ymax]}" -i uwbm_build.ini -d data_sources.yml -vvv
````
Oh the code is also in github: https://github.com/Deltares-research/urbanwb/blob/RTdeLange/urbanwb/read_parameter_base.py

But, also here we see a toml library

LuukBlom committed:
Update .github/workflows/tests.yml

Co-authored-by: Marnix <150045289+deltamarnix@users.noreply.github.com>

LuukBlom committed:
Update .github/workflows/tests.yml

Co-authored-by: Marnix <150045289+deltamarnix@users.noreply.github.com>

LuukBlom committed:
review comments:

- default config values for soiltype and croptype
- setup_landuse removed arguments: soiltype and croptype
- revert .toml to .ini (and to_toml to to_ini)
- rename UWBMConfig to UwbmConfig
- type hints

LuukBlom committed:
added sonar-project.properties file

LuukBlom committed:
remove redundant code

LuukBlom committed:
Update documentation

- Update installation instructions in README
- Update Quarto configuration
- Add new assets: logo and Deltares branding SVG.
- Fix all broken links in the docs
- Add how-to-build instructions for documentation.
- Update UWBM model class docstrings

LuukBlom committed:
pin hydromt to latest release

LuukBlom committed:
lint
